plugins/mission/mission.cpp

```diff
@@ -81,4 +95,101 @@ void Mission::subscribe_progress(progress_callback_t callback)
+        return Mission::Result::FAILED_TO_PARSE_QGC_PLAN;
+    }
+
+    // Compose mission items
+    Mission::Result result = [&items](const Json & mission_json) -> Mission::Result {
+        for (auto &mission_item : mission_json["items"].array_items())
+        {
+            // Parameters of Mission item & MAV command of it.
+            float speed_m_s = 0.f;
+            float gimbal_pitch_deg = 0.f, gimbal_yaw_deg = 0.f;
+            bool is_fly_through = false;
+            double lat_deg = NAN, lon_deg = NAN, rel_alt_deg = NAN;
+            size_t no_of_photos = 0;
+            auto camera_action = MissionItem::CameraAction::NONE;
+            auto command = mission_item["command"].int_value();
+
+            // Extract parameters of each mission item
+            std::vector<double> params;
+            for (auto &p : mission_item["params"].array_items()) {
+                params.push_back(p.number_value());
+            }
+
+            switch (command) {
+                case MAV_CMD_IMAGE_START_CAPTURE:
+                    no_of_photos = static_cast<size_t>(params[3]);
+                    if (no_of_photos > 1) {
+                        camera_action = MissionItem::CameraAction::START_PHOTO_INTERVAL;
+                    } else if (no_of_photos == 1) {
+                        camera_action = MissionItem::CameraAction::TAKE_PHOTO;
+                    }
+                    break;
+                case MAV_CMD_VIDEO_START_STREAMING:
+                    camera_action = MissionItem::CameraAction::START_VIDEO;
+                    break;
+                case MAV_CMD_VIDEO_STOP_STREAMING:
+                    camera_action = MissionItem::CameraAction::START_VIDEO;
+                    break;
+                case MAV_CMD_DO_CHANGE_SPEED:
+                    speed_m_s = params[2];
+                    break;
+                case MAV_CMD_DO_MOUNT_CONTROL:
+                    // Possible bug in QGroundcontrol. It stores -ve values for pitch in deg.
+                    gimbal_pitch_deg = -params[0];
+                    gimbal_yaw_deg = params[3];
+                    break;
+                case MAV_CMD_NAV_WAYPOINT:
+                    is_fly_through = (params[0] == 0.0) ? true : false;
+                case MAV_CMD_NAV_TAKEOFF:
+                    lat_deg = params[4];
+                    lon_deg = params[5];
+                    rel_alt_deg = params[6];
+                    break;
+                case MAV_CMD_NAV_RETURN_TO_LAUNCH:
+                    break;
+                default: // Unsupported mission command
+                    LogErr() << "Mission: Unsupported mission command: " << command;
+                    return Mission::Result::UNSUPPORTED_MISSION_CMD;
+            }
+            std::cout << "Cmd: " << command << " Lat: " << lat_deg << " Lon: " << lon_deg << " Alt: " <<
+                      rel_alt_deg <<
+                      " Speed " << speed_m_s << " Is fly thru: " << is_fly_through <<
```

Comment:
Does this print ` Is fly through: 1` ? Or `true` and `false`?

You could do: `<< (is_fly_through ? "yes" : "no") <<`

Reply:
Forgot! I will add.